app/src/main/java/org/mozilla/fenix/compose/tabstray/TabGridItem.kt

```diff
@@ -0,0 +1,336 @@
+                .padding(4.dp)
+                .combinedClickable(
+                    onLongClick = { onLongClick(tab) },
+                    onClick = { onClick(tab) }
+                ),
+            elevation = 0.dp,
+            shape = RoundedCornerShape(dimensionResource(id = R.dimen.tab_tray_grid_item_border_radius)),
+            border = BorderStroke(1.dp, FirefoxTheme.colors.borderPrimary)
+        ) {
+
+            Column(
+                modifier = Modifier.background(FirefoxTheme.colors.layer2)
+            ) {
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .wrapContentHeight()
+                ) {
+                    Favicon(
+                        url = tab.content.url,
+                        size = 16.dp,
+                        modifier = Modifier
+                            .align(Alignment.CenterVertically)
+                            .padding(
+                                8.dp, 0.dp, 0.dp, 0.dp
+                            )
+                    )
+
+                    FadingEndEdge(
+                        modifier = Modifier
+                            .weight(1f)
+                            .wrapContentHeight()
+                            .requiredHeight(30.dp)
+                            .padding(7.dp, 5.dp)
+                            .clipToBounds(),
+                        backgroundColor = FirefoxTheme.colors.layer2,
+                        isContentRtl = BidiFormatter.getInstance().isRtl(tab.content.title)
+                    ) {
+                        Text(
+                            text = tab.content.title,
+                            fontSize = 14.sp,
+                            maxLines = 1,
+                            softWrap = false,
+                            style = TextStyle(
+                                color = FirefoxTheme.colors.textPrimary,
+                                textDirection = TextDirection.Content
+                            )
+                        )
+                    }
+
+                    Icon(
+                        painter = painterResource(id = R.drawable.mozac_ic_close),
+                        contentDescription = stringResource(id = R.string.close_tab),
+                        tint = FirefoxTheme.colors.iconPrimary,
+                        modifier = Modifier
+                            .clickable { onCloseClick(tab) }
+                            .size(24.dp)
+                            .align(Alignment.CenterVertically)
+
+                    )
+                }
+                Divider(
+                    color = FirefoxTheme.colors.borderPrimary,
+                    thickness = 1.dp
+                )
+
+                Thumbnail(
+                    tab = tab,
+                    multiSelectionSelected = multiSelectionSelected,
+                )
+            }
+        }
+
+        MediaImage(
```

Comment:
Does this need to be hidden when `multiSelectionEnabled` is true, similar to `TabListItem`?